### PR TITLE
SqlQueueClient protection and watchdogs w/o connection factory

### PR DESCRIPTION
**Pull request by SergeyGaluzo**

### CODE REVIEW

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs [hunk 1]
```diff
@@ -23,37 +23,39 @@ namespace Microsoft.Health.Fhir.SqlServer.Features.Storage
...
+
+        private SqlQueueClient(ISqlRetryService sqlRetryService, ILogger<SqlQueueClient> logger)
+        {
+            _sqlRetryService = EnsureArg.IsNotNull(sqlRetryService, nameof(sqlRetryService));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+        }
+
+        public static SqlQueueClient GetInstance(ISqlRetryService sqlRetryService)
+        {
+            return new SqlQueueClient(sqlRetryService, new NullLogger<SqlQueueClient>());
```
Comment (fhibf) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs:48]: NullLogger? I think this class was only supposed to be used in tests, isn't it?

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs [hunk 2]
```diff
@@ -218,27 +219,48 @@ public async Task<IReadOnlyList<JobInfo>> GetJobsByIdsAsync(byte queueType, long
 
         public bool IsInitialized()
         {
+            if (_schemaInformation == null)
+            {
+                throw new InvalidOperationException("this method cannot be called with schema information = null");
```
Comment (fhibf) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs:224]: Low priority: Capital. **T**his

Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs]: Fixed

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs (hunk 1 above)
Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs]: Removed

**Review by fhibf — APPROVED** [review]

**Review by fhibf — APPROVED** [review]

**Pull request merged**